Ira/movies-app/src/components/homePage.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import Link from "react-router-dom";
+import Header from './header';
+import moviesData from './moviesData';
+import MovieCard from './movieCard'; 
+import NotFound from './notFound'; 
+import Layout from './layout';
+
+
+
+interface IHomePageState {
+   filteredMovies: Array<{}>;
+   searchInputValue: string;
+   searchBy: string;   
+   moviesCount: number;
+}
+
+class HomePage extends React.Component<IHomePageState> {   
+   
+   state = {
+      filteredMovies: [{}],   
+      searchInputValue: "",
+      searchBy: "title",
+      moviesCount: 0
+   }
+
+   changeSearchInputValue = (e: React.ChangeEvent<HTMLInputElement>) => {
+      this.setState({searchInputValue: e.target.value});
+   }
+   
+   setMoviesCount = () => {
+      this.setState({moviesCount: this.state.filteredMovies.length});
+   }
```

Comment:
I would suggest you not to create methods if you don't use it yet. In this case code will be more clear and will be less messy.